File: project/scripts/generate-translation-excel.py
import re
from openpyxl import Workbook
from openpyxl.styles import Font, PatternFill, Alignment, Border, Side, NamedStyle
from openpyxl.utils import get_column_letter

md_path = r'c:\Users\Mayn\CodeBuddy\20260407184558\translation-catalog.md'
output_path = r'c:\Users\Mayn\CodeBuddy\20260407184558\translation-catalog.xlsx'

with open(md_path, 'r', encoding='utf-8') as f:
    content = f.read()

lines = content.split('\n')

rows = []
current_source = None
current_namespace = None

for line in lines:
    stripped = line.strip()
    # Detect source sections
    if stripped.startswith('## 二、Web 端 UI 翻译'):
        current_source = 'web'
        current_namespace = None
        continue
    elif stripped.startswith('## 三、Mobile 端 UI 翻译'):
        current_source = 'mobile'
        current_namespace = None
        continue
    elif stripped.startswith('## 四、API 错误码消息'):
        current_source = 'api-error'
        current_namespace = None
        continue
    elif stripped.startswith('## 五、硬编码 Web 文本'):
        current_source = 'hardcoded-web'
        current_namespace = None
        continue
    elif stripped.startswith('## 六、硬编码 Mobile 文本'):
        current_source = 'hardcoded-mobile'
        current_namespace = None
        continue
    elif stripped.startswith('## 七、邮件模板文本'):
        current_source = 'email'
        current_namespace = None
        continue
    elif stripped.startswith('## '):
        current_source = None
        current_namespace = None
        continue

    if stripped.startswith('### '):
        current_namespace = stripped[4:].strip()
        continue

    if current_source and stripped.startswith('|') and 'Key' not in stripped and '---' not in stripped:
        parts = [p.strip() for p in stripped.split('|')]
        parts = [p for p in parts if p]
        if len(parts) >= 4:
            key = parts[0]
            zh = parts[1]
            en = parts[2]
            rows.append({
                'source': current_source,
                'namespace': current_namespace or '',
                'key': key,
                'zh': zh,
                'en': en,
            })
        elif len(parts) == 3 and current_source == 'api-error':
            # API error code tables have different format sometimes
            pass

# Parse API error codes separately since they use a slightly different format
api_section = re.search(r'## 四、API 错误码消息.*?(?=## 五、)', content, re.DOTALL)
if api_section:
    api_text = api_section.group(0)
    for line in api_text.split('\n'):
        stripped = line.strip()
        if stripped.startswith('|') and '错误码' not in stripped and '---' not in stripped:
            parts = [p.strip() for p in stripped.split('|')]
            parts = [p for p in parts if p]
            if len(parts) >= 3:
                key = parts[0]
                zh = parts[1]
                en = parts[2] if len(parts) > 2 else ''
                rows.append({
                    'source': 'api-error',
                    'namespace': 'errorCode',
                    'key': key,
                    'zh': zh,
                    'en': en,
                })

wb = Workbook()
ws = wb.active
ws.title = 'Translations'

# Headers
headers = ['来源(Source)', '分类(Namespace)', '键名(Key)', '中文(ZH)', '英文(EN)', '法语(FR)', '德语(DE)', '日语(JA)', '韩语(KO)', '备注(Context)']
ws.append(headers)

# Header style
header_fill = PatternFill(start_color='4472C4', end_color='4472C4', fill_type='solid')
header_font = Font(bold=True, color='FFFFFF', size=11)
header_align = Alignment(horizontal='center', vertical='center', wrap_text=True)
thin_border = Border(
    left=Side(style='thin'), right=Side(style='thin'),
    top=Side(style='thin'), bottom=Side(style='thin')
)

for col in range(1, len(headers) + 1):
    cell = ws.cell(row=1, column=col)
    cell.fill = header_fill
    cell.font = header_font
    cell.alignment = header_align
    cell.border = thin_border

# Alternate row fills
fill_even = PatternFill(start_color='D9E1F2', end_color='D9E1F2', fill_type='solid')
fill_odd = PatternFill(start_color='FFFFFF', end_color='FFFFFF', fill_type='solid')

for i, r in enumerate(rows, start=2):
    ws.append([
        r['source'],
        r['namespace'],
        r['key'],
        r['zh'],
        r['en'],
        '', '', '', '', ''
    ])
    row_fill = fill_even if i % 2 == 0 else fill_odd
    for col in range(1, len(headers) + 1):
        cell = ws.cell(row=i, column=col)
        cell.fill = row_fill
        cell.border = thin_border
        cell.alignment = Alignment(vertical='center', wrap_text=True)

# Column widths
ws.column_dimensions['A'].width = 18
ws.column_dimensions['B'].width = 22
ws.column_dimensions['C'].width = 35
ws.column_dimensions['D'].width = 40
ws.column_dimensions['E'].width = 45
ws.column_dimensions['F'].width = 40
ws.column_dimensions['G'].width = 40
ws.column_dimensions['H'].width = 40
ws.column_dimensions['I'].width = 40
ws.column_dimensions['J'].width = 30

# Freeze header
ws.freeze_panes = 'A2'

# Auto-filter
ws.auto_filter.ref = ws.dimensions

# Set row height for header
ws.row_dimensions[1].height = 30

wb.save(output_path)
print(f'Generated {output_path} with {len(rows)} rows')

File: project/scripts/generate-translation-excel.js
const fs = require('fs');
const XLSX = require('xlsx');

const mdPath = 'c:\\Users\\Mayn\\CodeBuddy\\20260407184558\\translation-catalog.md';
const outputPath = 'c:\\Users\\Mayn\\CodeBuddy\\20260407184558\\translation-catalog.xlsx';

const content = fs.readFileSync(mdPath, 'utf-8');
const lines = content.split(/\r?\n/);

const rows = [];
let currentSource = null;
let currentNamespace = null;

for (const line of lines) {
  const stripped = line.trimEnd();
  if (stripped.startsWith('## 二、Web 端 UI 翻译')) {
    currentSource = 'web';
    currentNamespace = null;
    continue;
  } else if (stripped.startsWith('## 三、Mobile 端 UI 翻译')) {
    currentSource = 'mobile';
    currentNamespace = null;
    continue;
  } else if (stripped.startsWith('## 四、API 错误码消息')) {
    currentSource = 'api-error';
    currentNamespace = null;
    continue;
  } else if (stripped.startsWith('## 五、硬编码 Web 文本')) {
    currentSource = 'hardcoded-web';
    currentNamespace = null;
    continue;
  } else if (stripped.startsWith('## 六、硬编码 Mobile 文本')) {
    currentSource = 'hardcoded-mobile';
    currentNamespace = null;
    continue;
  } else if (stripped.startsWith('## 七、邮件模板文本')) {
    currentSource = 'email';
    currentNamespace = null;
    continue;
  } else if (stripped.startsWith('## ')) {
    currentSource = null;
    currentNamespace = null;
    continue;
  }

  if (stripped.startsWith('### ')) {
    currentNamespace = stripped.slice(4).trim();
    continue;
  }

  if (currentSource && stripped.startsWith('|')) {
    const parts = stripped.split('|').map(p => p.trim());
    if (parts[0] === '') parts.shift();
    if (parts[parts.length - 1] === '') parts.pop();
    if (parts.length >= 3 && parts[0] !== 'Key' && parts[0] !== '错误码' && !parts[0].includes('---')) {
      rows.push([
        currentSource,
        currentNamespace || '',
        parts[0],
        parts[1],
        parts[2] || '',
        '', '', '', '', ''
      ]);
    }
  }
}

const wsData = [
  ['来源(Source)', '分类(Namespace)', '键名(Key)', '中文(ZH)', '英文(EN)', '法语(FR)', '德语(DE)', '日语(JA)', '韩语(KO)', '备注(Context)'],
  ...rows
];

const wb = XLSX.utils.book_new();
const ws = XLSX.utils.aoa_to_sheet(wsData);
XLSX.utils.book_append_sheet(wb, ws, 'Translations');

// Set column widths
ws['!cols'] = [
  { wch: 18 },
  { wch: 22 },
  { wch: 38 },
  { wch: 42 },
  { wch: 48 },
  { wch: 42 },
  { wch: 42 },
  { wch: 42 },
  { wch: 42 },
  { wch: 30 }
];

// Freeze first row
ws['!freeze'] = { xSplit: 0, ySplit: 1, topLeftCell: 'A2', activePane: 'bottomLeft' };

// Apply header styles via cell objects
const headerFill = { patternType: 'solid', fgColor: { rgb: '4472C4' } };
const headerFont = { bold: true, color: { rgb: 'FFFFFF' }, sz: 11 };
const headerAlign = { horizontal: 'center', vertical: 'center', wrapText: true };
const thinBorder = {
  left: { style: 'thin', color: { rgb: '000000' } },
  right: { style: 'thin', color: { rgb: '000000' } },
  top: { style: 'thin', color: { rgb: '000000' } },
  bottom: { style: 'thin', color: { rgb: '000000' } }
};

const evenFill = { patternType: 'solid', fgColor: { rgb: 'D9E1F2' } };
const oddFill = { patternType: 'solid', fgColor: { rgb: 'FFFFFF' } };

const range = XLSX.utils.decode_range(ws['!ref']);
for (let R = range.s.r; R <= range.e.r; ++R) {
  const isHeader = R === 0;
  const fill = isHeader ? headerFill : (R % 2 === 0 ? evenFill : oddFill);
  const font = isHeader ? headerFont : undefined;
  const alignment = isHeader ? headerAlign : { vertical: 'center', wrapText: true };

  for (let C = range.s.c; C <= range.e.c; ++C) {
    const cellRef = XLSX.utils.encode_cell({ r: R, c: C });
    if (!ws[cellRef]) ws[cellRef] = { v: '' };
    ws[cellRef].s = {
      fill,
      font,
      alignment,
      border: thinBorder
    };
  }
}

// Auto-filter
ws['!autofilter'] = { ref: ws['!ref'] };

XLSX.writeFile(wb, outputPath);
console.log(`Generated ${outputPath} with ${rows.length} rows`);
